src/box/box.cc

```diff
@@ -205,6 +205,20 @@ struct event *box_on_recovery_state_event;
 /** Cached event 'tarantool.trigger.on_change'. */
 static struct event *tarantool_trigger_on_change_event;
 
+/**
+ * Whether the given instance is configured to make bootstrap of
+ * the replicaset before instance UUID is assigned to the node.
+ *
+ * Has an effect only in the supervised bootstrap strategy.
+ *
+ * Has effect only on bootstrap, ignored afterwards. This way we
+ * don't have to clear the flag.
+ *
+ * Use bootstrap_leader_uuid to judge about an actual bootstrap
+ * leader.
+ */
+static bool am_i_bootstrap_leader;
```

Comment:
nit: as always, I have doubts about the naming) something like `is_forced_bootstrap_leader` or `is_configured_bootstrap_leader` sounds better to me, but is see that in https://github.com/tarantool/tarantool/blob/265c74763843f4a71613d592a884536c848f3b4d/src/box/lua/config/applier/box_cfg.lua#L747 you already used this name, so I guess I'm fine with it.

Reply:
I spin around naming for a while, but I didn't find anything better, to be honest.

'configured' sounds like something from the `box.cfg` table, but it is from a `box.ctl.<...>()` call.

'Forced' seems OK, though. OTOH, it doesn't look more forced than when we call `make_bootstrap_leader` during `box.cfg()`.

Maybe it is better to highlight that it is a kind of a message from `make_bootstrap_leader()`? `make_bootstrap_leader_result`, for example?

I would hear opinions here :)

Reply:
possibly `is_made_bootstrap_leader` or `is_user_made_bootstrap_leader`?

Reply:
> Let's just make is_bootsrap_leader not an argument in bootstrap() but a variable in replication.cc and set it from box.cc. This way replicaset_find_join_master will return NULL, when is_bootstrap_leader is set. No changes in bootstrap() will be needed.

Agreed.
Speaking about the name, it should be less general then. I think `is_made_bootstrap_leader` is a good one.

Probably `is_supervised_bootstrap_leader`?

Reply:
> Let's just make `is_bootsrap_leader` not an argument in `bootstrap()` but a variable in `replication.cc` and set it from `box.cc`.

If I understood you correctly, you propose to set it from `make_bootstrap_leader()` as well. I would prefer to don't do that, at least because we should take it into account only in the supervised bootstrap strategy. This way, we need separate 'leadership requested' and 'leadership is in effect' variables. In future, if we add some checks against two bootstrap leaders, there may be more situations, when the leadership request is refused.

Taking into account that I anyway need a separate `is_supervised_bootstrap_leader` variable, I wouldn't touch `is_bootsrap_leader`.

I like the `is_supervised_bootstrap_leader` name more than others. At least it explicitly states that it is related to the supervised bootstrap strategy.

I'm going to rename the variable, move it to `replication.{h,cc}`, take into account in `replicaset_find_join_master_supervised`.

Reply:
OTOH, if I anyway trying to keep the flag consistent, maybe it has sense to coalesce it with `is_bootsrap_leader`.

Will look around the relevant code, no sure.

Reply:
Renamed the flag to `is_supervised_bootstrap_leader`. I'm afraid of merging it with `is_bootstrap_leader` and more consistency questions you would ask me if I do :)